### PR TITLE
Modal component --closes #22

### LINKED ISSUE #22: Modal component
Modal:

modal

- should show / hide depending on `isOpen` prop,
- clicking outside the modal should close the modal,
- should work on mobile devices too (full-screen mode for mobile and close icon in the top right corner),

**Props:**
- `isOpen` - PropTypes.boolean, default to `false`,
- `header` - PropTypes.string, optional,
- `children` - PropTypes.node, required,

The idea is that this component can be used in following way:

```js
import { Modal } from 'unleash-styles'

<Modal isOpen={true} header={"Modal Test"}>
  <div>
    <p>This is just for testing purposes...</p>
  </div>
</Modal>
```

HINT: `this.props.children` will have this `<div>...</div>` assigned by default.

In the screenshot there is also an action button included, let's skip it for now and see the use-cases later on.

### PR DESCRIPTION
**Pull request by jacekelgda**

### CODE REVIEW

File: src/components/Modal/Modal.css [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.Modal {
+	position: fixed;
```
Comment (karolsojko) [inline src/components/Modal/Modal.css]: no indents ?

File: src/components/Modal/Modal.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  hideModal () {
+    this.setState({
+      isShowing: !this.state.isShowing
+    })
+  }
+
+  render () {
+    const modalClasses = classNames(
+      styles.Modal,
+      (!this.state.isShowing) && styles.isHidden
```
Comment (karolsojko) [inline src/components/Modal/Modal.js]: cool stuff 👍

File: src/components/Modal/Modal.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Modal component:
+
+    <ModalTrigger label="Open modal">
```
Comment (karolsojko) [inline src/components/Modal/Modal.md]: just a question - is `ModalTrigger` required to render `Modal`?

Comment (jacekelgda) [inline src/components/Modal/Modal.md]: No, I created this just to add to our living styleguides. It's an independent modal and needed to come up with some trigger to show it. Later, when we have floating button we can replace I think and show them together.

**Pull request merged**